### PR TITLE
Don't Log Any Messages on Non-IO MPI Ranks

### PR DESCRIPTION
**Pull request by bska**
This is a proof of concept to demonstrate one possible way of suppressing log messages on MPI ranks other than the IO rank (0).

Not intended for production.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

File: opm/simulators/utils/readDeck.cpp [hunk 1]
```diff
@@ -405,16 +419,27 @@ Opm::setupLogging(const int          mpi_rank_,
...
-        prtLog->setMessageFormatter(std::make_shared<Opm::SimpleMessageFormatter>(false));
+        if (mpi_rank_ == 0) {
+            std::shared_ptr<Opm::EclipsePRTLog> prtLog = std::make_shared<Opm::EclipsePRTLog>(logFileStream.str(), Opm::Log::NoDebugMessageTypes, false, output_cout_);
+            Opm::OpmLog::addBackend("ECLIPSEPRTLOG", prtLog);
+            prtLog->setMessageLimiter(std::make_shared<Opm::MessageLimiter>());
+            prtLog->setMessageFormatter(std::make_shared<Opm::SimpleMessageFormatter>(false));
+        }
+        else {
+            auto prtLog = std::make_shared<NullLog>(Opm::Log::NoDebugMessageTypes);
+            Opm::OpmLog::addBackend("ECLIPSEPRTLOG", std::move(prtLog));
```
Comment (blattms) [inline opm/simulators/utils/readDeck.cpp:430]: Would have used `move` here as I would assume it only saves an increment. Would rather skip the variable if there are performance concerns.

Comment (bska) [reply opm/simulators/utils/readDeck.cpp:430]: I appreciate the review, but just to be clear: This particular code will never be merged into the master branch.  It is only intended as a proof of concept.

**Comment by bska** [comment]
Closing this now as the proof-of-concept has served its purpose.

**Pull request closed without merging**